File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/GrabListener.java
package net.kdt.pojavlaunch;

/**
 * グラブ（カーソルキャプチャ）状態の変更を監視するためのリスナーインターフェース。
 */
public interface GrabListener {
    /**
     * グラブ状態が変更されたときに呼び出されます。
     * @param isGrabbing 現在グラブ中かどうか
     */
    void onGrabState(boolean isGrabbing);
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/view/GrabListener.kt
package com.arata.yukarilauncher.ui.view

/**
 * カーソルグラブ（捕捉）状態の変更を監視するための関数型インターフェース。
 * マウス操作がViewに捕捉されたときや解放されたときに通知を受け取ります。
 */
fun interface GrabListener {
    /**
     * グラブ状態が変更されたときに呼び出されます。
     * @param isGrabbing 現在グラブ中であればtrue
     */
    fun onGrabState(isGrabbing: Boolean)
}
